crates/primitives/receipt/src/to_starknet_types.rs

```diff
@@ -0,0 +1,203 @@
+    pub fn to_starknet_types(
+        self,
+        finality_status: starknet_types_rpc::TxnFinalityStatus,
+    ) -> starknet_types_rpc::DeployAccountTxnReceipt<Felt> {
+        starknet_types_rpc::DeployAccountTxnReceipt::<Felt> {
+            contract_address: self.contract_address,
+            common_receipt_properties: starknet_types_rpc::CommonReceiptProperties {
+                actual_fee: self.actual_fee.into(),
+                events: self.events.into_iter().map(starknet_types_rpc::Event::from).collect(),
+                execution_resources: self.execution_resources.into(),
+                finality_status,
+                messages_sent: self.messages_sent.into_iter().map(starknet_types_rpc::MsgToL1::from).collect(),
+                transaction_hash: self.transaction_hash,
+                execution_status: self.execution_result.into(),
+            },
+        }
+    }
+}
+
+impl From<FeePayment> for starknet_types_rpc::FeePayment<Felt> {
+    fn from(fee: FeePayment) -> Self {
+        Self { amount: fee.amount, unit: fee.unit.into() }
+    }
+}
+
+impl From<PriceUnit> for starknet_types_rpc::PriceUnit {
+    fn from(unit: PriceUnit) -> Self {
+        match unit {
+            PriceUnit::Wei => starknet_types_rpc::PriceUnit::Wei,
+            PriceUnit::Fri => starknet_types_rpc::PriceUnit::Fri,
+        }
+    }
+}
+
+impl From<MsgToL1> for starknet_types_rpc::MsgToL1<Felt> {
+    fn from(msg: MsgToL1) -> Self {
+        Self { from_address: msg.from_address, to_address: msg.to_address, payload: msg.payload }
+    }
+}
+
+impl From<Event> for starknet_types_rpc::Event<Felt> {
+    fn from(event: Event) -> Self {
+        Self {
+            from_address: event.from_address,
+            event_content: starknet_types_rpc::EventContent { keys: event.keys, data: event.data },
+        }
+    }
+}
+
+impl From<ExecutionResources> for starknet_types_rpc::ExecutionResources {
+    fn from(resources: ExecutionResources) -> Self {
+        Self {
+            bitwise_builtin_applications: nullify_zero(resources.bitwise_builtin_applications),
+            ec_op_builtin_applications: nullify_zero(resources.ec_op_builtin_applications),
+            ecdsa_builtin_applications: nullify_zero(resources.ecdsa_builtin_applications),
+            keccak_builtin_applications: nullify_zero(resources.keccak_builtin_applications),
+            memory_holes: nullify_zero(resources.memory_holes),
+            pedersen_builtin_applications: nullify_zero(resources.pedersen_builtin_applications),
+            poseidon_builtin_applications: nullify_zero(resources.poseidon_builtin_applications),
+            range_check_builtin_applications: nullify_zero(resources.range_check_builtin_applications),
+            segment_arena_builtin: nullify_zero(resources.segment_arena_builtin),
+            steps: resources.steps,
+            data_availability: resources.data_availability.into(),
+        }
+    }
+}
+
+fn nullify_zero(u: u64) -> Option<u64> {
```

Comment:
Consider making this a helper function inside of `impl From<ExecutionResources> for starknet_types_rpc::ExecutionResources` (nitpick)